CONTRIBUTING.md

```diff
@@ -178,19 +179,125 @@ This is usually the file you want to create to test certain compile behavior
+  - `compile_super_errors_ppx_v2(code: string)`: Compiles given code with the React v2 syntax
+  - `compile_super_errors_ppx_v3(code: string)`: Compiles given code with the React v3 syntax
+  - `load_module(cmi_path: string, cmi_content: string, cmj_name: string, cmj_content: string)`: Loads a module into the compiler (see notes on `cmj` / `cmi` below) 
+
+For each compile every successful operation will return `{js_code: string}`.
+On compile errors, the returned object will be `{js_error_msg: string}`.
+
+### Working on the Playground JS API
+
+Whenever you are modifying any files in the BuckleScript compiler, or in the `jsoo_main.ml` file, you'll need to rebuild the source and recreate the JS bundle.
+
+```
+node scripts/ninja.js config && node scripts/ninja.js build
+BS_PLAYGROUND=../playground node scripts/repl.js
+```
+
+**.cmj files in the Web**
+
+A `.cmj` file contains compile information and JS package information of
+BuckleScript build artifacts (your `.re / .ml` modules) and are generated on
+build (`scripts/ninja.js build`).
+
+A `.cmi` file is an [OCaml originated file
+extension](https://waleedkhan.name/blog/ocaml-file-extensions/) and contains all
+interface information of a certain module without any implementation.
+
+In this repo, these files usually sit right next to each compiled `.ml` / `.re`
+file. The structure of a `.cmj` file is defined in
+[js_cmj_format.ml](jscomp/core/js_cmj_format.ml). You can run a tool called
+`./jscomp/bin/cmjdump.exe [some-file.cmj]` to inspect the contents of given
+`.cmj` file.
+
+`.cmj` files are required for making BuckleScript compile modules (this includes
+modules like ReasonReact). BuckleScript includes a subset of modules by default,
+which can be found in `jscomp/stdlib-406` and `jscomp/others`. You can also find
+those modules listed in the `jsoo` call in `scripts/repl.js`. As you probably
+noticed, the generated `playground` files are all plain `.js`, so how are the `cmj` /
+`cmi` files embedded?
+
+`repl.js` calls an executable called `cmjbrowser.exe` on every build, which is a
+compile artifact from `jscomp/main/jscmj_main.ml`. It is used to serialize `cmj`
+/ `cmi` artifacts into two files called `jscomp/core/js_cmj_datasets.ml` and
+`jscomp/core/js_cmi_datasets.ml`. These files are only linked for the browser
+target, where BuckleScript doesn't have access to the filesystem. When working
+on BS, you'll see diffs on those files whenever there are changes on core
+modules, e.g. stdlib modules or when the ocaml version was changed. We usually
+check in these files to keep it in sync with the most recent compiler
+implementation. JSOO will pick up those files to encode them into the `exports.js`
+bundle.
+
+For any other dependency needed in the playground, such as `ReasonReact`, you
+will be required to serialize your `.cmi` / `.cmt` files accordingly so that
+JSOO's `ocaml.load` function can load the data. Right now we don't provide any
```

Comment:
`ocaml.load_module` , but this has nothing to do with `JSOO`. It's a function defined in `jsoo_main.ml`

Reply:
good catch